syntax = "proto3";

package TW.Bravo.Proto;
option java_package = "wallet.core.jni.proto";

// Input data necessary to create a signed transaction.
message SigningInput {
    // Chain id (256-bit number)
    bytes chain_id = 1;

    // Sender's username
    string sender = 2;

    // Recipient's username
    string recipient = 3;

    // Amount in "BRAVO" or "TEST"
    double amount = 4;

    // Whether the transaction is for testnet
    bool testnet = 5;

    // Memo to attach to the transaction
    string memo = 6;

    // Reference Block Id (160-bits)
    bytes reference_block_id = 7;

    // Timestamp on the reference block
    sfixed32 reference_block_time = 8;

    // Sender's private key's raw bytes
    bytes private_key = 9;
}

// Transaction signing output.
message SigningOutput {
    // JSON of the signed transaction.
    string json_encoded = 1;
}
